contracts/poap/src/state.rs

```diff
@@ -0,0 +1,123 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{Addr, Timestamp};
+use cw_storage_plus::{Item, Map};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Config {
+    pub admin: Addr,
+    pub minter: Addr,
+    pub mint_enabled: bool,
+    pub per_address_limit: u32,
+    pub cw721_code_id: u64,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct EventInfo {
+    pub creator: Addr,
+    pub start_time: Timestamp,
+    pub end_time: Timestamp,
+    pub base_poap_uri: String,
+    pub event_uri: String,
+}
+
+pub const CONFIG: Item<Config> = Item::new("config");
```

Comment:
Can we merge `CW721_ADDRESS` into `Config`? It would be useful to show the contract address.

Reply:
Actualy the cw721 address is returned in `QueryConfigResponse`,  I can mege CW721_ADDRESS and CONFIG btw.

Reply:
Oh, yeah. Just consider if let `QueryConfigResponse` returns config instead seperately fields.

Reply:
Yes I think it can be merged without problems @manu0466

Reply:
While performing the changes i noticed that this requires to store the cw721 address inside config as `Option` adding some boilerplate code to check that this field is not empty. So i prefer to keep them separate.

Reply:
@manu0466 Can it be empty address instead of `Option`? I am thinking about that the `instantiate message` should be successful with the proper `instantiate reply`, finally, the cw721 address must not empty after the reply. It can be checked with integration test in my opinion.

Reply:
I agree that the address will be stored since the instantiation is successful only if also the cw721 is properly instantiate but storing the address as empty string just to have an unique `Config` struct seems a dirt hack to me. What are the actual pros of this? We can have a the `QueryConfigResponse` returning just the `Config` struct but beside that i don't see anything else

Reply:
I am just thinking to reduce some code by merging it into config.  Agree with you, we should avoid dirt hack.